test_integration/test_ssl_verification.py

```diff
@@ -0,0 +1,62 @@
+# pylint: disable=missing-docstring
+import os
+import threading
+from http.server import HTTPServer, SimpleHTTPRequestHandler
+from ssl import PROTOCOL_TLS_SERVER, SSLContext
+
+import pytest
+from requests.exceptions import SSLError
+
+from ibm_cloud_sdk_core.base_service import BaseService
+from ibm_cloud_sdk_core.authenticators import NoAuthAuthenticator
+
+
+# The certificate files that are used in this tests are generated by this command:
+# openssl req \
+#    -new \
+#    -newkey rsa:4096 \
+#    -days 36500 \
+#    -nodes \
+#    -x509 \
+#    -subj "/C=US/CN=localhost" \
+#    -keyout test_ssl.key \
+#    -out test_ssl.cert
+
+
+def test_ssl_verification():
+    # Load the certificate and the key files.
+    cert = os.path.join(os.path.dirname(__file__), '../resources/test_ssl.cert')
+    key = os.path.join(os.path.dirname(__file__), '../resources/test_ssl.key')
+
+    # Build the SSL context for the server.
+    ssl_context = SSLContext(PROTOCOL_TLS_SERVER)
+    ssl_context.load_cert_chain(certfile=cert, keyfile=key)
+
+    # Create and start the server on a separate thread.
+    server = HTTPServer(('127.0.0.1', 3333), SimpleHTTPRequestHandler)
+    server.socket = ssl_context.wrap_socket(server.socket, server_side=True)
+    t = threading.Thread(target=server.serve_forever)
+    t.start()
+
+    # We run everything in a big try-except-finally block to make sure we always
+    # shutdown the HTTP server gracefully .
```

Comment:
```suggestion
    # shutdown the HTTP server gracefully.
```

Look what I just found!! @padamstx 😂

Reply:
oh gosh, how did we survive with that extra space? :joy: